Timeline View

spxrogers opened the pull request:
Added: custom list row for the time line as timeline_row.xml
Added: Timeline ListView configurations in MainActivity.
Added: Timline list on click listener to show detail view .. TODO: Change
placeholder Intent destination once detail view finished.
Modified: our Model API a little. Centralized access to Timeline object via
User-level abstraction.

spxrogers committed:
Added: custom list row for the time line as timeline_row.xml
Added: Timeline ListView configurations in MainActivity.
Added: Timline list on click listener to show detail view .. TODO: Change
placeholder Intent destination once detail view finished.
Modified: our Model API a little. Centralized access to Timeline object via
User-level abstraction.

spxrogers commented:
The last change: about the model api centralized to User, it felt more intuitive IMO to deal with user objects and adding and modifying data to them.

My motivations were: it seemed more organic to say "hey i'm using this and as a user i want to add a new event to my profile" and working inside our activity classes is much more succinct with `user.addEvent(theEvent);` instead of `user.getTimeline.getEvents().addEvent(theEvent);`

This is completely up for debate if you disagree though 

cc: @ZachSand @wibergln

ZachSand commented:
I like the changes with User, it seems more intuitive to me this way. The timeline view is a little unflattering and its not exactly obvious the textview rows are clickable pieces of content. Should we add a little '>' symbol to signify clicking the row does something?

spxrogers committed:
Added: Manifest read permissions.
Added: User verify permissions when selecting media.
Fixed: Inverted the UTimeline event sort order to show most recent first.
